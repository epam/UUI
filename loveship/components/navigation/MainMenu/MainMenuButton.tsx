import css from './MainMenuButton.scss';
import { Button, ButtonProps } from '@epam/uui-components';
import { IAdaptiveItem, withMods } from '@epam/uui-core';
import { ReactComponent as TriangleIcon } from '../../icons/chevron-down-24.svg';

export interface MainMenuButtonMods {
    type?: 'primary' | 'secondary';
}

export const MainMenuButton = withMods<ButtonProps, MainMenuButtonMods & IAdaptiveItem>(
    Button,
    mods => [css.root, css['type-' + (mods.type || 'primary')], css.fontSansSemibold],
    () => ({ dropdownIcon: TriangleIcon })
);
