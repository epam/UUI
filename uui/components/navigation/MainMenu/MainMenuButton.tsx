import * as React from 'react';
import { CX, cx, devLogger, IAdaptiveItem, Icon, IDropdownToggler, IHasCaption, IHasIcon, uuiElement } from '@epam/uui-core';
import { Clickable, ClickableComponentProps, IconContainer } from '@epam/uui-components';
import { CountIndicator } from '../../widgets';
import { ReactComponent as SvgTriangle } from '@epam/assets/icons/navigation-chevron_down-outline.svg';
import css from './MainMenuButton.module.scss';

interface MainMenuButtonMods {
    /**
    * Defines component type. The primary button leads to the main pages of the site, and the secondary to the others.
    */
    type?: 'primary' | 'secondary';
}

export type MainMenuButtonProps = MainMenuButtonMods & IAdaptiveItem & IDropdownToggler & Omit<ClickableComponentProps, 'isDisabled'>
& IHasIcon & IHasCaption & {
    /**
     * CSS classes to put on the caption
     * @deprecated
     * */
    captionCX?: CX;
    /** Icon for drop-down toggler */
    dropdownIcon?: Icon;
    /** Count value to be placed in component */
    count?: React.ReactNode;
};

export const MainMenuButton = /* @__PURE__ */React.forwardRef<HTMLButtonElement | HTMLAnchorElement | HTMLSpanElement, MainMenuButtonProps>((props, ref) => {
    if (__DEV__ && props.captionCX) {
        devLogger.warn('MainMenuButton: Property \'captionCX\' is deprecated and will be removed in the future release. Please use \'cx\' prop to access caption styles and use cascading to change the styles for the \'uui-caption\' global class');
    }

    const { type, ...clickableProps } = props;

    return (
        <Clickable
            { ...clickableProps }
            rawProps={ {
                role: 'menuitem',
                'aria-haspopup': props.isDropdown,
                'aria-expanded': props.isOpen,
                ...props.rawProps,
            } }
            cx={ [
                css.root,
                css['type-' + (type || 'primary')],
                props.cx,
            ] }
            ref={ ref }
        >
            { props.icon && props.iconPosition !== 'right' && (
                <IconContainer
                    icon={ props.icon }
                    onClick={ props.onIconClick }
                />
            ) }
            { props.caption && (
                <div className={ cx(uuiElement.caption, props.captionCX) }>
                    { props.caption }
                </div>
            ) }
            { props.count !== undefined && props.count !== null && (
                <CountIndicator
                    caption={ props.count }
                    color="neutral"
                    size="18"
                />
            ) }
            { props.icon && props.iconPosition === 'right' && (
                <IconContainer icon={ props.icon } onClick={ props.onIconClick } />
            ) }
            { props.isDropdown && (
                <IconContainer icon={ SvgTriangle } flipY={ props.isOpen } />
            )}
        </Clickable>
    );
});
