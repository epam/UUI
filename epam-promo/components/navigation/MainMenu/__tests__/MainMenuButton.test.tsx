import React from 'react';
import renderer from 'react-test-renderer';
import { MainMenuButton } from '../MainMenuButton';

describe('MainMenuButton', () => {
    it('should be rendered correctly', () => {
        const tree = renderer.create(<MainMenuButton />).toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('should be rendered correctly', () => {
        const tree = renderer
            .create(
                <MainMenuButton
                    caption="Test button"
                    type="primary"
                    target="_blank"
                    link={{ pathname: '/' }}
                    isDropdown
                    isOpen={false}
                    collapseToMore
                    estimatedWidth={120}
                    showInBurgerMenu
                    priority={6}
                />
            )
            .toJSON();
        expect(tree).toMatchSnapshot();
    });
});
