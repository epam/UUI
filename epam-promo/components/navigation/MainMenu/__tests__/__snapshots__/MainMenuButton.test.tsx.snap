// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainMenuButton should be rendered correctly 1`] = `
<button
  className="container uui-button-box uui-enabled root type-primary"
  href={null}
  onClick={[Function]}
  role="button"
  tabIndex={-1}
  type="button"
/>
`;

exports[`MainMenuButton should be rendered correctly 2`] = `
<a
  className="container uui-button-box uui-enabled -clickable root type-primary"
  onClick={[Function]}
  role="link"
  tabIndex={0}
  target="_blank"
>
  <div
    className="uui-caption"
  >
    Test button
  </div>
  <div
    className="container uui-icon uui-enabled"
    style={Object {}}
  >
    <svg
      className=""
    />
  </div>
</a>
`;
